### ISSUE TITLE
KeyNotFoundException with 2.4.1

### ISSUE THREAD

**Issue by rkg-mm** [CycloneDX/cyclonedx-dotnet#601]
Hi,

one of our projects builds is suddenly failing due to an exception in the tool. Is this a bug or what could be the source here? Was running fine until Monday (version 2.3.0 was used), now it seems a new version is out (2.4.1 used in next build automatically) its failing.

```
2022-10-13T08:27:59.8705975Z ##[section]Starting: Create BOM
[...]
2022-10-13T08:28:07.5407633Z Installing CycloneDX for NuGet...
2022-10-13T08:28:07.5408382Z You can invoke the tool using the following command: dotnet-CycloneDX
2022-10-13T08:28:07.5409934Z Tool 'cyclonedx' (version '2.4.1') was successfully installed.

2022-10-13T08:28:07.5412044Z Executing cycloneDX for NuGet...
2022-10-13T08:28:07.5412197Z 
2022-10-13T08:28:07.5412531Z Found the following local nuget package cache locations:
2022-10-13T08:28:07.5412905Z     /home/vsts/.nuget/packages/
2022-10-13T08:28:07.5413050Z 
[...]
2022-10-13T08:28:07.5429658Z Retrieving GitHub license for repository App-vNext/Polly and ref master
2022-10-13T08:28:07.5430209Z Retrieving GitHub license for repository dotnet/corefx and ref master
2022-10-13T08:28:07.5430693Z GitHub API failed with status code NotFound and message Not Found.
2022-10-13T08:28:07.5431168Z No license found on GitHub for repository dotnet/corefx using ref master
2022-10-13T08:28:07.5432127Z Unhandled exception. System.Collections.Generic.KeyNotFoundException: The given key 'CycloneDX.Models.NugetPackage' was not present in the dictionary.
2022-10-13T08:28:07.5432747Z    at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
2022-10-13T08:28:07.5433734Z    at CycloneDX.Program.<>c__DisplayClass117_0.<OnExecuteAsync>b__1(NugetPackage p) in /home/runner/work/cyclonedx-dotnet/cyclonedx-dotnet/CycloneDX/Program.cs:line 326
2022-10-13T08:28:07.5434448Z    at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
2022-10-13T08:28:07.5435014Z    at System.Collections.Generic.HashSet`1.UnionWith(IEnumerable`1 other)
2022-10-13T08:28:07.5435587Z    at System.Collections.Generic.HashSet`1..ctor(IEnumerable`1 collection, IEqualityComparer`1 comparer)
2022-10-13T08:28:07.5436188Z    at System.Linq.Enumerable.ToHashSet[TSource](IEnumerable`1 source, IEqualityComparer`1 comparer)
2022-10-13T08:28:07.5437195Z    at CycloneDX.Program.OnExecuteAsync(CommandLineApplication app) in /home/runner/work/cyclonedx-dotnet/cyclonedx-dotnet/CycloneDX/Program.cs:line 326
2022-10-13T08:28:07.5438006Z    at McMaster.Extensions.CommandLineUtils.Conventions.ExecuteMethodConvention.InvokeAsync(MethodInfo method, Object instance, Object[] arguments)
2022-10-13T08:28:07.5438806Z    at McMaster.Extensions.CommandLineUtils.Conventions.ExecuteMethodConvention.OnExecute(ConventionContext context, CancellationToken cancellationToken)
2022-10-13T08:28:07.5439565Z    at McMaster.Extensions.CommandLineUtils.Conventions.ExecuteMethodConvention.<>c__DisplayClass0_0.<<Apply>b__0>d.MoveNext()
2022-10-13T08:28:07.5440317Z --- End of stack trace from previous location ---
2022-10-13T08:28:07.5440880Z    at McMaster.Extensions.CommandLineUtils.CommandLineApplication.ExecuteAsync(String[] args, CancellationToken cancellationToken)
2022-10-13T08:28:07.5441648Z    at McMaster.Extensions.CommandLineUtils.CommandLineApplication.ExecuteAsync[TApp](CommandLineContext context, CancellationToken cancellationToken)
2022-10-13T08:28:07.5442630Z    at CycloneDX.Program.Main(String[] args) in /home/runner/work/cyclonedx-dotnet/cyclonedx-dotnet/CycloneDX/Program.cs:line 130
2022-10-13T08:28:07.5443199Z    at CycloneDX.Program.<Main>(String[] args)
```

Environment:
Azure Pipeline using following .NET sdk to run the tool:
```
Tool to install: .NET Core sdk version 6.x.
Found version 6.0.402 in channel 6.0 for user specified version spec: 6.x
Version: 6.0.402 was found in cache.
```

Thanks in advance!

**Comment by BlythMeister** [comment]
I have raised a PR to fix this in #600

**Comment by rkg-mm** [comment]
> I have raised a PR to fix this in #600

Thank you!
